R/generate_dist.R

```diff
@@ -595,3 +601,27 @@
     nTrt, balanced, strata, grpName, ratio
   )[, get(grpName)]
 }
+
+# Internal function called by .generate - returns cluster size data
+
+.genclustsize <- function(dtSim, totalss, dispersion = 0,  
+                          n = nrow(dtSim), envir) { 
+  
+    total <- .evalWith(totalss, .parseDotVars(totalss, envir))
+    d <- .evalWith(dispersion, .parseDotVars(dispersion, envir))
+  
+    if (d == 0) {
+      ss <- floor(total/n)
+      s <- rep(ss, n)
+      surplus <- total - ss*n
+      if (surplus > 0) {
+        upgrade <- sample(1:n, surplus, replace = FALSE)
+        s[upgrade] <- s[upgrade] + 1
+      }
+    } else {
+      x <- dirmult::rdirichlet(1, alpha = rep(1/d, n) )[1,]
```

Comment:
You added `dirmult` as a Suggests so you should guard this function to error gracefully if the package is not installed (I think we have done this in other functions too?)